plugin/src/main/scala/chisel3/internal/plugin/ChiselComponent.scala

```diff
@@ -171,22 +180,34 @@ class ChiselComponent(val global: Global, arguments: ChiselPluginArguments)
       // Check if a subtree is a candidate
       case dd @ ValDef(mods, name, tpt, rhs) if okVal(dd) =>
         val tpe = inferType(tpt)
+        val isData = shouldMatchData(tpe)
+
         // If a Data and in a Bundle, just get the name but not a prefix
-        if (shouldMatchData(tpe) && inBundle(dd)) {
+        if (isData && inBundle(dd)) {
           val str = stringFromTermName(name)
           val newRHS = transform(rhs) // chisel3.internal.plugin.autoNameRecursively
           val named = q"chisel3.internal.plugin.autoNameRecursively($str)($newRHS)"
           treeCopy.ValDef(dd, mods, name, tpt, localTyper.typed(named))
         }
         // If a Data or a Memory, get the name and a prefix
-        else if (shouldMatchNamedComp(tpe)) {
+        else if (isData || shouldMatchNamedComp(tpe)) {
           val str = stringFromTermName(name)
           // Starting with '_' signifies a temporary, we ignore it for prefixing because we don't
           // want double "__" in names when the user is just specifying a temporary
           val prefix = if (str.head == '_') str.tail else str
           val newRHS = transform(rhs)
           val prefixed = q"chisel3.experimental.prefix.apply[$tpt](name=$prefix)(f=$newRHS)"
-          val named = q"chisel3.internal.plugin.autoNameRecursively($str)($prefixed)"
+
+          val named =
+            if (isData || !shouldMatchChiselPrefixed(tpe))
+              // Generate autoNameRecursively call if:
+              // - the type is a data, or
+              // - the type is not a AffectsChiselPrefix
+              q"chisel3.internal.plugin.autoNameRecursively($str)($prefixed)"
+            else
+              // Do not generate if the type is both not a data and a AffectsChiselPrefix
+              prefixed
```

Comment:
```suggestion
          val named =
            if (isNamedComp) {
              // Only name named components (not things that are merely prefixed)
              q"chisel3.internal.plugin.autoNameRecursively($str)($prefixed)"
            } else {
              prefixed
            }
```
1. Always use curly braces on multi-line if-else, we don't want to be the next [Goto fail](https://en.wikipedia.org/wiki/Unreachable_code#goto_fail_bug). We can probably make Scalafmt do this and we really should.
2. The comments were a little redundant so I shortened them, the code does basically say what it is doing
3. I minimized the number of lookups with a couple of extra vals above, see other comment